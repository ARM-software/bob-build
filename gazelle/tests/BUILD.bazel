load("@bazel_gazelle//:def.bzl", "gazelle_generation_test")

[gazelle_generation_test(
    name = file[0:-len("/WORKSPACE")],
    size = "small",
    gazelle_binary = "//:gazelle_bin",
    test_data = glob(
        include = [file[0:-len("/WORKSPACE")] + "/**"],
    ),
) for file in glob(["**/WORKSPACE"])]
