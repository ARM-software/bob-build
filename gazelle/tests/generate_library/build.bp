// Main build file
bob_library {
    name: "lib_forward_defines",
    srcs: ["src/libs/lib.cpp"],
    hdrs: ["src/libs/liba.h"],
    feature_a: {
        hdrs: ["src/libs/libb.h"],
    },
    includes: ["internal/include"],
    local_defines: ["LOCAL_DEFINE"],
    defines: ["FORWARDED_DEFINE"],
    host_supported: true,
}

bob_library {
    name: "lib_dep",
    deps: ["lib_forward_defines"],
    host_supported: true,
    target_supported: false,
    build_by_default: true,
}

bob_library {
    name: "lib_alwayslink",
    srcs: ["libfoo.cc"],
    alwayslink: true,
}

bob_library {
    name: "lib_empty_do_not_generate",
    alwayslink: true,
}
