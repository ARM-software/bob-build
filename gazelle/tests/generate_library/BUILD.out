load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")

# gazelle:bob_root

config_setting(
    name = "[__FEATURE_A]",
    flag_values = {"//:FEATURE_A": True},
)

config_setting(
    name = "![FEATURE_A]",
    flag_values = {"//:FEATURE_A": False},
)

alias(
    name = "[FEATURE_A]",
    actual = ":[__FEATURE_A]",
)

bool_flag(
    name = "FEATURE_A",
    build_setting_default = True,
)

cc_library(
    name = "lib_forward_defines",
    srcs = ["src/libs/lib.cpp"],
    hdrs = ["src/libs/liba.h"] + select({
        "//conditions:default": [],
        "//:[FEATURE_A]": ["src/libs/libb.h"],
    }),
    defines = ["FORWARDED_DEFINE"],
    local_defines = ["LOCAL_DEFINE"],
)

cc_library(
    name = "lib_dep",
    deps = ["//:lib_forward_defines"],
)

cc_library(
    name = "lib_alwayslink",
    srcs = ["libfoo.cc"],
    alwayslink = True,
)
